drivers/spi/spi.go

```diff
@@ -0,0 +1,44 @@
+package spi
+
+import (
+	"golang.org/x/exp/io/spi"
+	"sync"
+	"golang.org/x/exp/io/spi/driver"
+	"fmt"
+)
+
+type Connection struct {
+	connection *spi.Device
+	address int
+	mutex *sync.Mutex
+}
+
+// NewConnection creates and returns a new connection to a specific
+// spi device on a bus and address
+func NewConnection(opener driver.Opener, address int) (connection *Connection) {
+	dev, err := spi.Open(opener)
+	if err != nil {
+		panic(err)
+	}
+
+	return &Connection{connection: dev, address: address, mutex: &sync.Mutex{}}
+}
+
+// TODO: Read - the exp/io/spi implementation doesn't currently support reading
```

Comment:
I really like what's going on here overall!

So regarding adding the spi interfaces to Gobot...

The way that this would best fit into Gobot's architecture, is by defining an interface that Gobot uses for spi. 

Different Gobot adaptors would then implement this interface. For example the raspi platform could use the x/io/spi interface to provide its implementation. The firmata platform would add additional calls to the appropriate Firmata sysex calls, etc.

Then any Gobot spi drivers would work with any Gobot adaptor that provides the Gobot spi interface. The GoPiGo is obviously design just for the raspi platform, but other Gobot spi drivers would be able to work as described here.

Does this make sense?